frontend/src/entry/model/settings/EntryRolesInfo.tsx

```diff
@@ -37,17 +52,32 @@ export default function EntryRolesInfo({ entry }: EntryRolesInfoProps) {
 
   const schemaRolesList = useMemo(() => getFilteredRoles(RoleKind.SCHEMA), [getFilteredRoles])
 
+  const showPopover = (row: Row) => {
```

Comment:
`showPopover` and `canRoleDoAction` are just components written as normal functions. Consider doing the following (but feel free to keep them in the same file as they're very specific to `EntryRolesInfo`):
```
type RowHelpTextProps = {
  row: Row
}

function RowHelpText({ row }: RowHelpTextProps) {
  if (!row.helpText) {
    return null
  }

  return <HelpPopover>{row.helpText}</HelpPopover>
}

type RoleActionIconProps = {
  role: RoleValue
  roles: RoleValue[]
}

function RoleActionIcon({ role, roles }: RoleActionIconProps) {
  if (!roles.includes(role)) {
    return null
  }

  return <Done />
}
```
They would be used like so:
```
          <Stack direction='row' alignItems='center'>
            {row.permission}
            <RowHelpText row={row} />
          </Stack>
```
```
        <TableCell align='center'>
          <RoleActionIcon role={RoleValue.OWNER} roles={row.roles} />
        </TableCell>
```